sotodlib/qa/metrics.py

```diff
@@ -0,0 +1,77 @@
+class QAMetric(object):
+
+    # these will be defined by child classes
+    _influx_meas = None  # measurement to record to
+    _influx_field = None  # the field to populate
+
+    def __init__(self, context, monitor, log="qa_metrics_log"):
+        """ A QA metric base class.
+
+        Arguments
+        ---------
+        context : core.Context
+            Context that includes all necessary metadata to generate metrics.
+        monitor : site_pipeline.monitor.Monitor
+            InfluxDB connection.
+        log : str
+            InfluxDB measurement that is used to log new entries.
+        """
+
+        self.context = context
+        self.monitor = monitor
+        self._influx_log = log
+
+    def exists(self, obs_id):
+        """ Check if a metric exists for this obs_id in Influx.
+
+        Arguments
+        ---------
+        obs_id : str
+            The observation ID to check.
+
+        Returns
+        -------
+        exists : bool
+            Whether it exists or not.
+        """
+        return self.monitor.check(self._influx_field, obs_id, log=self._influx_log)
```

Comment:
hm, this may not be unique enough, unless I log each measurement separately. Still figuring out how this is supposed to work.